specification/machinelearningservices/data-plane/Microsoft.MachineLearningServices/preview/2019-05-01/hyperdrive.json

```diff
@@ -0,0 +1,714 @@
+              "$ref": "#/definitions/HyperDriveCancelExperimentResponse"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/HyperDriveErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Cancel a HyperDrive experiment": {
+            "$ref": "./examples/cancelHyperDriveExperiment.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "HyperDriveErrorResponse": {
+      "properties": {
+        "code": {
+          "format": "int32",
+          "type": "integer",
+          "description": "Error response code."
+        },
+        "message": {
+          "type": "string",
+          "description": "Error message."
+        }
+      },
+      "required": [
+        "code",
+        "message"
+      ],
+      "type": "object",
+      "description": "Response in case of an error."
+    },
+    "HyperDriveCancelExperimentResponse": {
+      "properties": {
+        "code": {
+          "format": "int32",
+          "type": "integer",
+          "description": "Response code."
+        },
+        "result": {
+          "type": "string",
+          "description": "Message indicating operation success."
+        }
+      },
+      "required": [
+        "code",
+        "result"
+      ],
+      "type": "object",
+      "description": "Response for HyperDrive_CancelExperiment in case of success."
+    },
+    "HyperDriveExperimentBase": {
+      "description": "Base object for both request and response of HyperDrive_CreateExperiment api.",
+      "properties": {
+        "description": {
+          "maxLength": 511,
+          "type": "string",
+          "x-nullable": true,
+          "description": "The description for Hyperdrive run."
+        },
+        "generator_config": {
+          "description": "Hyperparameter space and the sampling method configuration.",
+          "properties": {
+            "name": {
+              "type": "string",
+              "enum": [
+                "RANDOM",
+                "GRID",
+                "BAYESIANOPTIMIZATION"
+              ],
+              "description": "Hyperparameter sampling method.",
+              "x-ms-enum": {
+                "name": "Name",
+                "modelAsString": true
+              }
+            },
+            "parameter_space": {
+              "type": "object",
+              "additionalProperties":{
+                "type": "array",
+                "items": {},
+                "minItems": 2,
+                "maxItems": 2,
+                "description": "The first element in the array is a string identifying the parameter expression type like choice, uniform etc. The second element is a parameter expressions corresponding to the type. Please refer '#/definitions/ChoiceParameterExpression', '#/definitions/UniformParameterExpression' etc.",
+                "example": ["choice", [[1, 2, 3, 4]]]
```

Comment:
Is this example correct? Is the second element supposed to be an array of arrays?

Reply:
This is one of the possible options. "example": ["uniform", [0, 1]] will also work. But for "choice" we  expect this to be array of arrays as example shows in line 537.

Reply:
Postponed. Captured work for next preview.